src/cli/cmd/mod.rs

```diff
@@ -127,4 +131,22 @@ impl FlakeHubClient {
 
         Ok(orgs)
     }
+
+    async fn versions(&self, flake: String, constraint: String) -> Result<Vec<Version>, FhError> {
+        let version = urlencoding::encode(&constraint);
+
+        let endpoint = self
+            .api_addr
+            .join(&format!("versions/{}/{}", flake, version))?;
```

Comment:
Can we start using [`Url::path_segments_mut`](https://docs.rs/url/latest/url/struct.Url.html#method.path_segments_mut) for constructing these URLs? String-concatenation for URLs has proved problematic in the past.

Reply:
Mostly because it's quite annoying but yeah, I should fix this

Reply:
Okay, I think I've cleaned this up pretty substantially